recipes/blast/2.2.31/build.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -euxo pipefail
+
+cd $SRC_DIR/c++/
+
+export CFLAGS="$CFLAGS -O2"
+export CXXFLAGS="$CXXFLAGS -O2 -std=c++14"
+export CPPFLAGS="$CPPFLAGS -I$PREFIX/include"
+export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
+export CC_FOR_BUILD=$CC
+
+if test x"`uname`" = x"Linux"; then
+    # only add things needed; not supported by OSX ld
+    LDFLAGS="$LDFLAGS -Wl,-as-needed"
+fi
+
+if [ `uname` == Darwin ]; then
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib -lz -lbz2"
+else
+    export CPP_FOR_BUILD=$CPP
+fi
+
+LIB_INSTALL_DIR=$PREFIX/lib/ncbi-blast+
+
+# with/without options:
+#
+# dll: enable dynamic linking
+# mt: enable multi-threading
+# -autodep: no automatic dependency build (one time build)
+# -makefile-auto-update: no rebuild of makefile (one time build)
+# flat-makefile: use single makefile
+# -caution: disable configure script warnings
+# -dbapi: don't build database connectivity libs
+# -lzo: don't add lzo support
+# runpath: set runpath for installed $PREFIX location
+# hard-runpath: disable new dtags (disallow LD_LIBRARY_PATH override on Linux)
+# -debug: disable debug
+# strip: remove debugging symbols (size!)
+# z: set zlib
+# bz2: set libbz2
+# -openssl: disable openssl
+# -gcrypt: disable gcrypt (needed on OSX)
+# -krb5: disable kerberos (needed on OSX)
+
+# Fixes building on Linux
+export AR="${AR} rcs" 
+
+./configure \
+    --with-dll \
+    --with-mt \
+    --with-openmp \
+    --without-autodep \
+    --without-makefile-auto-update \
+    --with-flat-makefile \
+    --without-caution \
+    --without-dbapi \
+    --without-lzo \
+    --with-hard-runpath \
+    --with-runpath=$LIB_INSTALL_DIR \
+    --without-debug \
+    --with-strip \
+    --with-z=$PREFIX \
+    --with-bz2=$PREFIX \
+    --without-krb5 \
```

Comment:
```suggestion
    --without-boost \
    --without-krb5 \
```
Should fix the OSX build, and the linux build doesn't use boost either.
This is the OSX build:
```
09:23:38 BIOCONDA INFO (ERR) usage: basename string [suffix]
09:23:38 BIOCONDA INFO (ERR)        basename [-a] [-s suffix] string [...]
09:23:38 BIOCONDA INFO (ERR) configure: WARNING: Untested Boost version; may prove incompatible.
09:23:38 BIOCONDA INFO (ERR) configure: WARNING: If so, please re-run this script with the flag --without-boost.
09:23:45 BIOCONDA INFO (ERR) /Users/distiller/project/miniconda/conda-bld/blast_1575393586052/work/c++/src/build-system/configure: line 38894: cd: UNKNOWN: No such file or directory
```
The linux build doesn't use boost either:
```
16:30:12 BIOCONDA INFO (OUT) checking Boost version...
16:30:12 BIOCONDA INFO (OUT) checking for Boost library name tag... -gcc73-mt--64 (autodetected)
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Regex... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Spirit... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.System... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Filesystem... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Iostreams... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Program-Options... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Test... no
16:30:12 BIOCONDA INFO (OUT) checking for Boost.Threads... no
```

Reply:
Or we find out which boost version is required and pin it.

Reply:
The newest blast builds also don't include boost, that might be out of scope ? what's your take on that @dpryan79 ?

Reply:
Given how old this version of blast is it's unlikely to require a boost version that would be compatible with other things in the same environment. I guess for Galaxy that's not a big deal though. I wouldn't personally bother.